packages/flutter_goldens_client/lib/skia_client.dart

```diff
@@ -0,0 +1,198 @@
+  Future<bool> auth(Directory workDirectory) async {
+    _workDirectory = workDirectory;
+
+    if (_serviceAccount == null)
+      return false;
+
+    final String authorization ='${_workDirectory.path}serviceAccount.json';
+    await io.File(authorization).writeAsString(_serviceAccount);
+
+    final List<String> authArguments = <String>[
+      'auth',
+      '--service-account', authorization,
+      '--work-dir', _workDirectory.childDirectory('temp').path,
+    ];
+
+    final io.ProcessResult authResults = io.Process.runSync(
+      _goldctl,
+      authArguments,
+    );
+
+    if(authResults.exitCode != 0) {
+      final StringBuffer buf = StringBuffer();
+      buf
+        ..writeln('Flutter + Skia Gold auth failed.')
+        ..writeln('stdout: ${authResults.stdout}')
+        ..writeln('stderr: ${authResults.stderr}');
+      throw NonZeroExitCode(authResults.exitCode, buf.toString());
+    }
+    return true;
+  }
+
+  Future<void> imgtestInit() async {
+    final String commitHash = await _getCommitHash();
+    final String keys = '${_workDirectory.path}keys.json';
+    final String failures = '${_workDirectory.path}failures.json';
+
+    await io.File(keys).writeAsString(_getKeysJSON());
+    await io.File(failures).create();
+
+    final List<String> imgtestInitArguments = <String>[
+      'imgtest', 'init',
+      '--instance', 'flutter',
+      '--work-dir', _workDirectory
+        .childDirectory('temp')
+        .path,
+      '--commit', commitHash,
+      '--keys-file', keys,
+      '--failure-file', failures,
+      '--passfail',
+    ];
+
+    if (imgtestInitArguments.contains(null)) {
+      final StringBuffer buf = StringBuffer();
+      buf.writeln('Null argument for Skia Gold imgtest init:');
+      imgtestInitArguments.forEach(buf.writeln);
+      throw NonZeroExitCode(1, buf.toString());
+    }
+
+    final io.ProcessResult imgtestInitResult = io.Process.runSync(
+      _goldctl,
+      imgtestInitArguments,
+    );
+
+    if (imgtestInitResult.exitCode != 0) {
+      final StringBuffer buf = StringBuffer();
+      buf
+        ..writeln('Flutter + Skia Gold imgtest init failed.')
+        ..writeln('stdout: ${imgtestInitResult.stdout}')
+        ..writeln('stderr: ${imgtestInitResult.stderr}');
+      throw NonZeroExitCode(imgtestInitResult.exitCode, buf.toString());
+    }
+  }
+
+  Future<bool> imgtestAdd(String testName, File goldenFile) async {
+    final List<String> imgtestArguments = <String>[
+      'imgtest', 'add',
+      '--work-dir', _workDirectory.childDirectory('temp').path,
+      '--test-name', testName.split('.png')[0],
+      '--png-file', goldenFile.path,
+    ];
+
+    if(imgtestArguments.contains(null)) {
+      final StringBuffer buf = StringBuffer();
+      buf.writeln('Null argument for Skia Gold imgtest add:');
+      imgtestArguments.forEach(buf.writeln);
+      throw NonZeroExitCode(1, buf.toString());
+    }
+
+    //final io.ProcessResult imgtestResult =
```

Comment:
probably best to remove the commented out code (maybe just leave the TODO). The code will just bit-rot and not work anymore.